packages/ember-core/src/components/form/search.gts

```diff
@@ -302,7 +310,7 @@ export default class Search<T> extends BoundValue<
     evt.preventDefault();
     evt.stopPropagation();
 
-    this.visibility.show(evt);
+    this.searchString = this.displayValue;
```

Comment:
This should really be done in the `onBlur`, otherwise you get some weird behavior when selecting a value.

https://github.com/user-attachments/assets/48d60eb1-280e-492a-a21c-bb44052c96eb